Change detector panics in develop after fix

```shell
=== FAIL: tests/integration/query/one_to_one TestQueryOneToOneWithIdFieldOnPrimary (0.03s)
2023-05-24T22:54:10.519-0400, INFO, tests.integration, One-to-one relation primary direction, id field with name clash on primary side, {"Database": "badger-file-system"}
2023-05-24T22:54:10.539-0400, INFO, db, Closing DefraDB process...
2023-05-24T22:54:10.548-0400, INFO, db, Successfully closed running process
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x60 pc=0x100f2ec68]

goroutine 4490 [running]:
testing.tRunner.func1.2({0x1015602c0, 0x10222fbd0})
	/usr/local/go/src/testing/testing.go:1396 +0x1c8
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1399 +0x378
panic({0x1015602c0, 0x10222fbd0})
	/usr/local/go/src/runtime/panic.go:884 +0x204
github.com/sourcenetwork/defradb/tests/integration.getDocuments({0x1017a4e70, 0x14000046090}, 0x140008feb70?, {{0x101100f5a, 0x4f}, {0x140008feb70, 0x3, 0x3}}, {0x1400020a1e0, 0x1, ...}, ...)
	/private/var/folders/5j/5mpqq_mj103gyyb21sd1smr00000gn/T/defradb/tests/changeDetector/code/20e9286eee169fb2186b6ad00d648818b7373564/tests/integration/utils2.go:806 +0x288
github.com/sourcenetwork/defradb/tests/integration.executeTestCase({0x1017a4e70, 0x14000046090}, 0x140004c2820, {0x14000869e70, 0x2, 0x2}, {{0x101100f5a, 0x4f}, {0x140008feb70, 0x3, ...}}, ...)
	/private/var/folders/5j/5mpqq_mj103gyyb21sd1smr00000gn/T/defradb/tests/changeDetector/code/20e9286eee169fb2186b6ad00d648818b7373564/tests/integration/utils2.go:316 +0x254
github.com/sourcenetwork/defradb/tests/integration.ExecuteTestCase(0x10167d3e0?, {0x14000869e70, 0x2, 0x2}, {{0x101100f5a, 0x4f}, {0x140008feb70, 0x3, 0x3}})
	/private/var/folders/5j/5mpqq_mj103gyyb21sd1smr00000gn/T/defradb/tests/changeDetector/code/20e9286eee169fb2186b6ad00d648818b7373564/tests/integration/utils2.go:283 +0x354
github.com/sourcenetwork/defradb/tests/integration/query/one_to_one.TestQueryOneToOneWithIdFieldOnPrimary(0x1400024d200?)
	/private/var/folders/5j/5mpqq_mj103gyyb21sd1smr00000gn/T/defradb/tests/changeDetector/code/20e9286eee169fb2186b6ad00d648818b7373564/tests/integration/query/one_to_one/with_id_field_test.go:116 +0x220
testing.tRunner(0x140004c2820, 0x101789358)
	/usr/local/go/src/testing/testing.go:1446 +0x10c
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1493 +0x300
```

--- a/docs/data_format_changes/i1530-change-detector-without-non-mutations.md
+++ b/docs/data_format_changes/i1530-change-detector-without-non-mutations.md
@@ -1,0 +1,3 @@
+# Change detector without non-mutations actions
+
+The previous fix caused a regression in the change detector and we need a documentation to break the cycle.

--- a/tests/integration/utils2.go
+++ b/tests/integration/utils2.go
@@ -549,8 +549,6 @@ ActionLoop:
 		} else if firstNonSetupIndex > -1 {
 			// -1 to exclude this index
 			endIndex = firstNonSetupIndex - 1
-		} else {
-			startIndex = endIndex
 		}
 	} else {
 		if setupCompleteIndex > -1 {
@@ -560,6 +558,7 @@ ActionLoop:
 			// We must not set this to -1 :)
 			startIndex = firstNonSetupIndex
 		} else {
+			// if we down't have any non-mutation actions, just use the last action
 			startIndex = endIndex
 		}
 	}

--- a/tests/integration/utils2.go
+++ b/tests/integration/utils2.go
@@ -558,7 +558,7 @@ ActionLoop:
 			// We must not set this to -1 :)
 			startIndex = firstNonSetupIndex
 		} else {
-			// if we down't have any non-mutation actions, just use the last action
+			// if we don't have any non-mutation actions, just use the last action
 			startIndex = endIndex
 		}
 	}

